src/base/AsyncCallQueue.cc

```diff
@@ -15,52 +15,25 @@
 
 AsyncCallQueue *AsyncCallQueue::TheInstance = 0;
 
-AsyncCallQueue::AsyncCallQueue(): theHead(NULL), theTail(NULL)
-{
-}
-
-void AsyncCallQueue::schedule(AsyncCall::Pointer &call)
-{
-    assert(call != NULL);
-    assert(!call->theNext);
-    if (theHead != NULL) { // append
-        assert(!theTail->theNext);
-        theTail->theNext = call;
-        theTail = call;
-    } else { // create queue from cratch
-        theHead = theTail = call;
-    }
-}
+AsyncCallQueue::AsyncCallQueue() = default;
```

Comment:
I would move this to the header.

Reply:
Yes, please. Such a move should avoid wasting more time on this issue. The point of a `= default` declaration in a .cc file is to hide the (generated) implementation from the header users. In some cases, such hiding solves dependency problems and/or reduces code size.

In this particular case, we are hiding how the `scheduled` data member is initialized, but the gain is probably very small (if any) -- the current code including AsyncCallQueue.h may not change when we stop hiding because that code "knows too much" already. Moreover, for proper hiding, we may need to hide the generated destructor implementation as well, and we do not do that.

Reply:
Removed.